<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>pATLAS</title>

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/black.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
      link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
		<style>
			.alignLeft{
				text-align: left;
				font-size: 20pt !important;
			}
			.alignLeftBig{
				text-align: left;
				font-size: 30pt !important;
			}
			.reveal h2 .customHeader{
				color: red;
			}
		</style>
	</head>
	<body>
	<div class="reveal">
		<div class="slides">
			<section data-background-image="images/patlas_expansion.gif"
							 data-background-size="contain"
							 data-background-color="white"
							style="color: black">
					<!--<textarea data-template>-->
				<img src="images/pATLAS_black.png" style="width: 30%; border:none;
				box-shadow: none;"/>
				<h2 style="color: black"> pATLAS: the hitchhikers guide to the plasmid
					galaxy</h2>

				Jesus TF, Ribeiro-Gonçalves B, Silva DN, Valeria B, Ramirez M,
					Carriço JA
					<!--</textarea>-->
			</section>
			<section>
				<section>
					<h2 style="color: #3487ff">Plasmids</h2>
					<ul style="list-style-type:disc">
						<li class="alignLeftBig">Extrachromosomal linear or circular DNA
						</li>
						<ul style="list-style-type:square">
							<li class="alignLeftBig">Mosaicism (between strains, species
								and genera)</li>
							<li class="alignLeftBig">Highly modular and chimeric</li>
						</ul>
					</ul>

					<img src="images/plasmid1.png" style="background-color: white; width: 30%">
				</section>
				<section>
					<h2 style="color: #3487ff">Plasmids</h2>
					<ul style="list-style-type:disc">
						<li class="alignLeftBig">Mobile Elements
						</li>
						<li class="alignLeftBig">Key components in horizontal gene transfer
						</li>
						<ul style="list-style-type:square">
							<li>They often carry genes that may benefit
								the survival of the organism (e.g antibiotic resistance)</li>
						</ul>
					</ul>
					<img src="images/Conjugation.svg.png" style="background-color: white; width: 40%">
				</section>
			</section>
			<section>
				<section>
					<!--<textarea data-template>-->
					<h2 style="color: #3487ff">Why</h2>
					<ul style="list-style-type:disc">
						<li class="alignLeftBig">Available tools (disclaimer: these
							are <b>not</b> people's names)</li>
						<ul style="list-style-type:square">
							<li class="alignLeftBig">Assembly-based</li>
							<ul style="list-style-type:circle">
								<li class="alignLeftBig">cBar</li>
								<li class="alignLeftBig">PlasmidFinder</li>
								<li class="alignLeftBig">plasmidSPAdes</li>
								<li class="alignLeftBig">PlasmidTron</li>
								<li class="alignLeftBig">Recycler</li>
							</ul>
						</ul>
					</ul>
					<!--</textarea>-->
				</section>
				<section data-markdown>
						<textarea data-template>
							<h2 style="color: #3487ff">Why</h2>

* Available tools (disclaimer: these are **not** people's names)
	* Mapping-based
		* PlasmidProfiler (based on PlasmidFinder database)
					</textarea>
				</section>
			</section>
			</section>
			<section>
				<section data-markdown>

					<textarea data-template>
						<h2 style="color: #3487ff">We present pATLAS</h2>

* NCBI refseq database - **11.667** sequences
	* Featuring **10.949** plasmid named sequences and some oddities!
	* Featuring weird species names (e.g _Enterobacteriaceae bacterium_)

* Pairwise comparisons using [MASH](https://github.com/marbl/Mash) to construct
a network of plasmid relationships
	* MASH distance of 0.1 ~ ANI 0.9
	* More than **110 Million** pairwise comparisons.
						</textarea>
				</section>
				<section data-markdown>

					<textarea data-template>
						<h2 style="color: #3487ff">We present pATLAS</h2>

* [Abricate](https://github.com/tseemann/abricate)
	* PlasdmidFinder (plasmid families)
	* ResFinder and Card databases (resistance genes)
	* VFDB (virulence genes)

* [Flask](http://flask.pocoo.org/) app (back-end).
* Html + css + javascript (front-end)
	* [Vivagraph.js](https://github.com/anvaka/VivaGraphJS) (force-directed graph)
	* [Highcharts.js](https://www.highcharts.com/) (for some exportable plots)

					</textarea>
				</section>
				<section data-markdown>

					<textarea data-template>
						<h2 style="color: #3487ff">We present pATLAS</h2>

* Auxiliary scripts (run locally):
	* [PlasmidCoverage](https://github.com/tiagofilipe12/PlasmidCoverage):
		* Mapping (**bowtie2** and **samtools**)
	* [mash_wrapper](https://github.com/tiagofilipe12/mash_wrapper):
		* Containment in reads (**Mash screen**)
		* Calculate distance of a sequence (**Mash dist**)

* Outputs (JSON format) can be loaded into pATLAS.
						</textarea>
				</section>
				<section data-markdown>

					<textarea data-template>
						<h2 style="color: #3487ff">We present pATLAS</h2>

* Available to users in [www.patlas.site](http://www.patlas.site).

* Source code at [pATLAS Github](https://github.com/tiagofilipe12/pATLAS).
  * Please feel free to open issues and make suggestions.

					</textarea>
        </section>
      </section>
      <section>
        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul style="list-style-type:disc">
            <li class="alignLeftBig">Initial visualization:
            </li>
            <img src="images/initial_viz.gif" style="width: 100%; border:none;
				box-shadow: none;"/>
          </ul>
        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul style="list-style-type:disc">
            <li class="alignLeftBig">Plasmid information:
            </li>
            <img src="images/plasmid_click.gif" style="width: 100%; border:none;
				box-shadow: none;"/>
          </ul>
        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul style="list-style-type:disc">
            <li class="alignLeftBig">Browse by taxa:
            </li>
            <img src="images/taxa_search.gif" style="width: 100%; border:none;
				box-shadow: none;"/>
          </ul>
        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul style="list-style-type:disc">
            <li class="alignLeftBig"><b>Other metadata:</b>
              <ul style="list-style-type:square">
                <li class="alignLeftBig"><b>Distance</b> - mash distance between two
                  plasmids</li>
                <li class="alignLeftBig"><b>Length</b> - The size in bp of the
                  plasmids</li>
                <li class="alignLeftBig"><b>Size ratio</b> - The difference in size
                  between two plasmids</li>
              </ul>
            </li>
          </ul>
          <img src="images/distance_result.png" style="width: 60%; border:none;
				box-shadow: none; "/>
        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul style="list-style-type:disc">
            <li class="alignLeftBig">Browse by resistance genes:
            </li>
            <img src="images/browse_resistance.gif" style="width: 100%; border:none;
				box-shadow: none;"/>
          </ul>
        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Browse</h2>

          <ul lass="alignLeftBig" style="list-style-type:disc">
            <li class="alignLeftBig"><b>Other annotations:</b>
              <ul style="list-style-type:square">
                <li class="alignLeftBig"><b>Plasmid families</b> - PlasmidFinder
                </li>
                <li class="alignLeftBig"><b>Virulence</b> - VFDB
                </li>
              </ul>
            </li>
            <img src="images/browse_pf.png" style="position: relative;
            left: 10%; bottom: 0%; width: 80%; border:none; box-shadow: none; "
            />
          </ul>

        </section>

      </section>

      <section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Import</h2>

          <ul lass="alignLeftBig" style="list-style-type:disc">
            <li class="alignLeftBig"><b>Mapping results</b>
            </li>
            <li class="alignLeftBig"><b>Mash screen results</b>
            </li>
            <li class="alignLeftBig"><b>Assembly/Mash dist screen results</b>
            </li>
            <img src="images/import_menu.png" style="position: relative;
            left: 24%; bottom: 0%; width: 52%; border:none; box-shadow: none; "
            />
          </ul>

        </section>

        <section data-markdown>
          <textarea data-template>
          <h2 style="color: #0074d7">pATLAS - pATLASflow</h2>

* A series of pipelines:
  * Using [NextFlow](https://www.nextflow.io/) (the easy way):
     * Easier to provide inputs
     * More capable to manage multiple inputs (e.g. multiple pairs of reads)
     * No need to install third party programs manually (e.g. bowtie2)
  * Using python scripts (the hard way)
          </textarea>

        </section>

        <section>
          <h2 style="color: #0074d7">pATLAS - Import</h2>

          <ul lass="alignLeftBig" style="list-style-type:disc">
            <li class="alignLeftBig"><b>Results example</b>
            </li>
            <img src="images/import_mapping.gif" style="width: 100%; border:none; box-shadow: none; "
            />
          </ul>

        </section>

      </section>

      <section>

        <section>
					<h2 style="color: #0074d7">pATLAS - Statistics</h2>

					<ul class="alignLeftBig" style="list-style-type:disc">
						<li class="alignLeftBig"><b>Taxa</b>
						</li>
						<li class="alignLeftBig"><b>Length</b>
						</li>
						<li class="alignLeftBig"><b>Annotations</b>
						</li>
						<img src="images/plots.gif" style="border:none;
						box-shadow: none"/>
					</ul>




        </section>

      </section>

      <section data-markdown>

					<textarea data-template>
						<h2 style="color: #3487ff">Conclusions</h2>

* User-friendly interface to browse NCBI refseq plasmid database
	* Allows to discover annotation issues
	* Allows to check for conflicts between CARD and ResFinder
	* Clear separation between Gram+ and Gram-
	* Robust clustering of Gram+ bacteria (by genera)
* This method can be used with any sequence type (e.g. TE and ICEs)

					</textarea>
      </section>
      </section>
      <section data-markdown data-background-color="white">
					<textarea data-template>
						<h2>Acknowledgements</h2>

						<li class="alignLeft">FCT - "Fundação para a Ciência e a
							Tecnologia" and	BacGenTrack (TUBITAK/0004/2014)</li>

						<li class="alignLeft">Andrew McArthur</li>
						<li class="alignLeft">Adam Phillippy</li>
						<li class="alignLeft">Catarina Inês Mendes</li>
						<li class="alignLeft">Natacha Couto</li>
						<img src="images/UMMIbioinfo.jpg" style="width: 30%; position:
						absolute; bottom: 35%; right: 1%"/>
						<img src="images/mramirez.jpg" style="width: 10%; position:
						absolute; bottom: 35%; right: 33%"/>
						<img src="images/dns.png" style="width: 10%; position:
						absolute; bottom: 54%; right: 33%"/>
						<img src="images/ummi1.png" style="width: 18%; position:
						relative; bottom: 2%; right: 2%"/>
						<img src="images/The_Logo_of_Tübitak.jpg" style="width: 10%;
						position: relative; bottom: 2%; right: 2%"/>
						<img src="images/2015_FCT_V_color.jpg" style="width: 20%; position:
						relative; bottom: 2%; right: 2%"/>
						<img src="images/imm.png" style="width: 20%; position:
						relative; bottom: 2%; right: 2%"/>
					</textarea>
				</section>

			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.initialize({
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad() } }
				]
			})
		</script>
	</body>
</html>
